std/math.d

```diff
@@ -2532,74 +2532,161 @@ unittest
  *      $(TR $(TD $(NAN))            $(TD FP_ILOGBNAN) $(TD no))
  *      )
  */
-int ilogb(real x)  @trusted nothrow @nogc
+int ilogb(T)(const T x) @trusted nothrow @nogc
+    if(isFloatingPoint!T)
 {
-    version (Win64_DMD_InlineAsm)
+    import core.bitop : bsr;
+    // Provide a bsr implementation for ulong in 32-bit mode
+    int bsr_ulong(ulong x) @trusted pure nothrow @nogc
```

Comment:
@ibuclaw How sure are you that the bsr64 in std.numeric works for both little endian and big endian? Using a union instead of bitshifts makes a large assumption about memory layout that I am not sure is justified.